app/src/main/cpp/ControllerContainer.cpp

```diff
@@ -318,6 +342,130 @@ ControllerContainer::GetControllers() const {
   return m.list;
 }
 
+template <typename GenericVector>
+static bool loadHandMeshAttribute(tinygltf::Model& model,
+                                  int accessorIndex, int expectedType, int expectedComponentType,
+                                  size_t typeByteSize, int numComponents,
+                                  std::vector<GenericVector>& vector) {
+  auto& accessor = model.accessors[accessorIndex];
+
+  if (accessor.type != expectedType || accessor.componentType != expectedComponentType)
+    return false;
+  if (accessor.bufferView >= model.bufferViews.size())
+    return false;
+  auto& bufferView = model.bufferViews[accessor.bufferView];
+  if (bufferView.buffer >= model.buffers.size())
+    return false;
+  auto& buffer = model.buffers[bufferView.buffer];
+
+  vector.resize(accessor.count);
+  memcpy(vector.data(), buffer.data.data() + bufferView.byteOffset + accessor.byteOffset,
+         accessor.count * typeByteSize * numComponents);
+
+  return true;
+}
+
+bool ControllerContainer::LoadHandMeshFromAssets(Controller& aController) {
+  auto& handMesh = aController.handMesh;
+
+  tinygltf::TinyGLTF modelLoader;
+  tinygltf::Model model;
+  std::string err;
+  std::string warn;
+  bool ok = modelLoader.LoadBinaryFromFile(&model, &err, &warn,
+                                           aController.leftHanded ? "hand-model-left.glb" : "hand-model-right.glb");
+  if (!ok) {
+    VRB_ERROR("Error loading hand mesh asset: %s", err.c_str());
+    return false;
+  }
+  if (!warn.empty()) {
+    VRB_WARN("%s", warn.c_str());
+  }
+
+  if (model.meshes.size() == 0)
+    return false;
+  // Assume the first mesh
+  auto& mesh = model.meshes[0];
+
+  if (mesh.primitives.size() == 0)
+    return false;
+  // Assume the first primitive of the mesh
+  auto& primitive = mesh.primitives[0];
+
+  // Load indices
+  if (primitive.indices < 0 || primitive.indices >= model.accessors.size())
+    return false;
+  if (!loadHandMeshAttribute(model, primitive.indices,
+                             TINYGLTF_TYPE_SCALAR, TINYGLTF_COMPONENT_TYPE_UNSIGNED_SHORT,
+                             sizeof(uint16_t), 1, handMesh.indices)) {
+    return false;
+  }
+  handMesh.indexCount = handMesh.indices.size();
+
+  // Load vertex attributes
+  for (auto& attr: primitive.attributes) {
+    if (attr.second >= model.accessors.size())
+      return false;
+
+    bool error = false;
```

Comment:
I do prefer using here the opposite of error and not negate the output of `loadHandMeshAttribute` calls which is a bit weird.

Reply:
Agree, but the problem is that there could be more attributes that we shall ignore, so we have to explicitly set the variable to positive. It is a bit more code, but fair enough.